splash/browser_tab.py

```diff
@@ -503,6 +505,29 @@ def runjs(self, js_source, handle_errors=True):
         js_source = "%s;undefined" % js_source
         self.evaljs(js_source, handle_errors=handle_errors)
 
+    def runjs_async(self, js_source, callback, errback, timeout=0):
+        """
+        An async variant of runjs.
+
+        It injects a callback function named <lua_resume> into the scope of
+        the user's script. When lua_resume() is called, the corresponding
+        Lua coroutine will resume.
```

Comment:
I originally called it `evaljs_async`, but I changed it as I was working, because I think `evaljs` implies that we evaluate an expression (or group of expressions) and return the value of that expression (or value of the last expression). On the other hand, `runjs` does not care about any expressions. I agree it is perhaps confusing to overload the meaning of these terms.

`wait_for_return` makes sense to me. Also, what about `wait_for_callback`, `wait_for_resume`, or `yield_to_js`?